test_user_management.py

```
@@ -0,0 +1,391 @@
+        except Exception as e:
+            logging.error(f"Error creating user: {e}")
+            return False
+    
+    def authenticate_user(self, username: str, password: str) -> Optional[Dict]:
+        """Authenticate user with username and password"""
+        try:
+            conn = sqlite3.connect(self.db_path)
+            cursor = conn.cursor()
+            
+            # SECURITY ISSUE 4: SQL injection vulnerability
+            query = f"SELECT * FROM users WHERE username = '{username}' AND password = '{password}'"
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)